src/btree/bt_read.c

```diff
@@ -715,8 +722,7 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
             /*
              * Check if the page requires forced eviction.
              */
-            if (evict_skip || LF_ISSET(WT_READ_NO_SPLIT) || btree->evict_disabled > 0 ||
-              btree->lsm_primary)
```

Comment:
Is it OK to move these flags (and the new check) up to the initialization of evict_skip? That local variable gets overwritten at other places in the loop here, so it seems possible that moving the flag checking to the initialization will change the behavior.

Reply:
I don't see any way we can get into trouble, the things we're checking here don't change in the function (and the only other action on `evict_skip` in the function is to set it to `true`).

That said, I originally moved the code because the test was complex (with all the `WT_SESSION.name` strcmp calls. Now that it's simpler, I'll move it back to where it was just for simplicity of the change.

Pushed in acc79033f.